src/webpack-angular-types-plugin/type-extraction/type-extraction.ts

```diff
@@ -15,13 +24,70 @@ function getterOrSetterInputExists(
     return !!entity && entity.kind === "input";
 }
 
+/**
+ * Creates mappings from generic class arguments of a subClass to the generic
+ * type parameters of the parent class. References are stored in weakMaps using
+ * ts-morph symbols (which are actually classes) to be able to resolve encountered
+ * types in a class by reference.
+ *
+ * Example, given the following two classes:
+ *  class ChildComponent extends ParentComponent<string, number>{}
+ *  class ParentComponent<S, T> {}
+ *
+ * The resulting mapping would look something like this:
+ * WeakMap {
+ *     <Symbol of ParentComponent> => WeakMap {
+ *          <Symbol of S> => string
+ *          <Symbol of T> => number
+ *     }
+ *}
+ *
+ * This information can later be used to map properties in base-classes to the
+ * actual type that is available when creating type-docs for a subClass.
+ */
+function extractGenericTypesFromClass(
+    classDeclarations: ClassDeclaration[]
+): GenericTypeMapping {
+    const result = new WeakMap<TsMorphSymbol, Type>();
+    for (const classDeclaration of classDeclarations) {
+        const targetTypeArguments = classDeclaration
+            .getExtends()
+            ?.getTypeArguments()
+            .map((a) => a.getType());
+        if (targetTypeArguments && targetTypeArguments.length > 0) {
```

Comment:
Can be shortened

```
targetTypeArguments?.length > 0)
```

Reply:
This does not seem to work, because `targetTypeArguments` can be undefined, and `undefined > 0` is an invalid operation.
I significantly simplified the method anyways, so this check was removed.